src/mds/MDSRank.cc

```diff
@@ -2983,11 +2983,71 @@ bool MDSRankDispatcher::handle_command(
 
     damage_table.erase(id);
     return true;
+  } else if (prefix == "drop cache") {
+    command_drop_cache(ds);
+    return true;
   } else {
     return false;
   }
 }
 
+void MDSRank::command_drop_cache(stringstream *ds)
+{
+  if (not is_active()) {
+    dout(5) << __func__ << ": MDS not active, no-op" << dendl;
+    return;
+  }
+
+  if (not mdcache->trim(UINT64_MAX))
+    return;
+
+  server->recall_client_state(1, true);
+  mds_lock.Unlock();
+  if (not _wait_for_client_recall_state_acks())
+    dout(4) << "Timed out; didn't not receive "
+	    "CEPH_SESSION_RECALL_STATEMSG_ACK from all sessions." << dendl;
+
+  // Optionally, flush the journal -
+  Formatter *fmtr = new JSONFormatter(true);
```

Comment:
Use:

`std::unique_ptr<Formatter> fmtr(new JSONFormatter(true));`

then

`fmtr.reset(new JSONFormatter(true));`

when you want to create a new one.